lib/Browser.js

```diff
@@ -131,8 +132,18 @@ class Browser extends EventEmitter {
    * @return {!Promise<string>}
    */
   async version() {
-    const version = await this._connection.send('Browser.getVersion');
-    return version.product;
+    if (!this._version)
```

Comment:
nit: since this is not a hot path or anything I'd avoid caching to keep the program state small.

Reply:
https://github.com/GoogleChrome/puppeteer/pull/1716/commits/a48d759035777c22a509c4b8a61c29b20d6b6874